packages/cardano-services/src/Asset/TypeOrmNftMetadataService.ts

```diff
@@ -10,15 +10,23 @@ export class TypeOrmNftMetadataService extends TypeormService implements NftMeta
 
   async getNftMetadata(assetInfo: AssetPolicyIdAndName): Promise<Asset.NftMetadata | null> {
     const assetId = Cardano.AssetId.fromParts(assetInfo.policyId, assetInfo.name);
-    const stringAssetName = Buffer.from(assetInfo.name, 'hex').toString('utf8');
     return this.withDataSource(async (dataSource) => {
       const queryRunner = dataSource.createQueryRunner();
-      const nftMetadataRepository = queryRunner.manager.getRepository(NftMetadataEntity);
+      let asset: NftMetadataEntity | null;
 
-      const asset = await nftMetadataRepository.findOneBy({
-        name: stringAssetName,
-        userTokenAsset: { id: assetId }
-      });
+      try {
+        const nftMetadataRepository = queryRunner.manager.getRepository(NftMetadataEntity);
+        asset = await nftMetadataRepository
+          .findOneBy({
+            userTokenAsset: { id: assetId }
+          })
+          .catch((error) => {
+            this.logger.error(error);
+            return null;
+          });
+      } finally {
```

Comment:
This would be a little cleaner
```suggestion
      } catch (error) {
        this.logger.error(error);
      } finally {
```